includes/data-port/class-sensei-data-port-job.php

```diff
@@ -373,15 +374,29 @@ public function run() {
 				$has_processed_task = true;
 			}
 
+			if ( ! $task->is_completed() ) {
```

Comment:
This minor refactor should prevent the need for an extra run at the end of a job.